qemu_guest_agent: Support get vfio nvme disk info

YvanY0 opened the pull request:
Add "vfio_nvme" variant into "check_get_disks" and "check_get_diskstats". this depends on #3800 which provides a decorator to set up host devices.

ID: 1222

PaulYuuu commented:
Depends on:
1. https://github.com/avocado-framework/avocado-vt/pull/3744
2. https://github.com/avocado-framework/avocado-vt/pull/3750
3. https://github.com/autotest/tp-qemu/pull/3800

PaulYuuu commented:
@6-dehan feel free to review this patch, thanks. cc @vivianQizhu

Commit:
qemu_guest_agent: Support get vfio nvme disk info

Add "vfio_nvme" variant into "check_get_disks" and
"check_get_diskstats". This is depends on PR 3800 which provide a
context manager to set up host devices.

Signed-off-by: Yihuang Yu <yihyu@redhat.com>
